### PR TITLE
Week 3 assessment al

### PR DESCRIPTION
**Pull request by alappies**

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -15,46 +15,146 @@
...
 // Example input: 10
 // Expected output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
 
 
-
 // b) Create the function that makes the test pass.
+//Psuedo Coding
+//First I need to create a function 
+// I need the function to take in a number and return it as an array 
+//I need the array to spit out the fibonacci sequence starting from 2 with the end point being the number I input.
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:36]: this is a good start, explain "How" you want this to happen.

File: code-challenges.test.js [hunk 2]
```diff
@@ -15,46 +15,146 @@
...
 
+// var num1 = 1
+// const fibonacciMe = (input) =>{
+//     let newArr = []  
+//     for (let index = 0; index < input.length; index++) {
+//     return newArr((num1 -1) + (num1 -2))
+//     }
+//     return newArr.push(fibonacciMe)
+
+// }
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:54]: Okay both of these solutions are close! 
- What issues did you run into? 
- What responses did you get? 
- Did you get the tests to pass? 

This is all necessary information if others are going to be able to help you.

File: code-challenges.test.js [hunk 3]
```diff
@@ -15,46 +15,146 @@
...
 
+// var fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
+// Expected output: [-823, 7, 23]
 
 // b) Create the function that makes the test pass.
-
-
+//Pseudo Code:
+// I need a function that takes in arrays  const sortThisOut  = [] ==>{}
+// I need that function to return only numbers using typeOf()
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:83]: This is great! why didn't you implement this?

File: code-challenges.test.js [hunk 4]
```diff
@@ -15,46 +15,146 @@
...
+// }
+// return newArray.typeOf(Number)
+// }
+// // console.log(sortThisOut(fullArr2));
+
+describe(sortThisOut, ()=> {
+    it('takes in an array and returns a new array of only odd numbers sorted from least to greatest', ()=> {
+    expect(sortThisOut(fullArr1)).toEqual([-9, 7, 9, 199])
+    })
+    })
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:115]: Lets focus on communicating those responses and test results! You got this!

File: code-challenges.test.js [hunk 5]
```diff
@@ -15,46 +15,146 @@
...
+    it('is function that takes in an array and returns an array of the accumulating sum', ()=> {
+    expect(iCannot(numbersToAdd1)).toEqual([2, 6, 51, 60])
+    })
+    })
+
+// //Psuedo Codding:
+// First I need to make a function that takes in an array
+// I need to much like the fibonacci sequence have the array go through a for loop
+// I need to iterate the numbers in the array through an equation that will add the current number to the number of the previous index. 
+// I need it to return in the form of an array.
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:159]: Okay I can see here you mostly copied the fib sequence math. That can sometimes throw us off really hard as developers. Pseudo coding this problem out in your own words and writing down all the logical steps will make this problem a lot easier on yourself.

File: interview-questions.md [hunk 6]
```diff
@@ -10,58 +10,72 @@ Then, research the question to expand on your answer. Even if you feel you have
 1. What is `this` in JavaScript?
 
   Your answer:
+  The relavance of `this` is apparent in the Javacript method of "class." You would use "this" when refering to something you want to add a value to in your class. example: this.gas = 10.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:13]: This is a solid answer!

File: interview-questions.md [hunk 7]
```diff
@@ -10,58 +10,72 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
   Researched answer:
-
+  "this" is in refrence to an "object" in the class method in javascript, and as per usual, I  was only half right above, an object is the cross of data and function so it gains a value and a function.
 
 
 2. What is React? Why would you use it?
 
   Your answer:
+  This I feel like I can confidently answer without much research. React is a mulitfacited open-source tool you can use to build user interfaces such as webpages and apps. It can be used to build decadent websites like "Netflix" and detailed apps like facebook and twitter. I would use React because it is one of the leading industry tools in app developement today and once you get used to the JSX language, it can be pretty straight forward to use (realize I did not say easy).
```
Comment (WalkerTexas-coder) [inline interview-questions.md:22]: Amazing answer!

File: interview-questions.md [hunk 8]
```diff
@@ -10,58 +10,72 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 3. Which lifecycle method is required in a React class component?
 
   Your answer:
+  The lifecycle method that is required to use in a React class component is "State".
 
   Researched answer:
-
+The constructor is the method required for a react class component.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:35]: Almost, it's the render() method! Honestly would've gotten me on this one too.

File: interview-questions.md [hunk 9]
```diff
@@ -10,58 +10,72 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
   Your answer:
+  When you run yarn in your react, there is a large file called "node-modules" and "yarn.lock" that are nested right into your program. I'm not sure what files are "modified" when you run yarn, but I know that you are making the majority of the modifications to "src" folder and adding components to that folder, so if I had to make an educated guess, it would be that it modified the "src." 
 
   Researched answer:
-
+Ok so this was a trick question. "yarn" is a javascript packaging manager that makes up a react app, so when you run yarn you open up a browser address called "local host 3000" and you're hosting your own react website/app.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:54]: correct everything that we use in terminal with a word in the front (yarn, git, node) all reefers to a software that we are calling commands from.

File: react-debugging.html [hunk 10]
```diff
@@ -79,6 +77,7 @@ <h1>Draw a Card</h1>
                   )
                 })}
               </div>
+              </>
```
Comment (WalkerTexas-coder) [inline react-debugging.html:80]: Great work!

**Review by WalkerTexas-coder — APPROVED** [review]
Ashley, you have some brilliant interview answers and your react debugging is on point! Where I feel this assessment missed on an opportunity was in your pseudo code. You have some solid logic in your code but without the communication that explains where you're headed and whats going on with the code its almost impossible to know when you've gotten what you want. 

For instance:  
- Did any of your tests pass? 
- What kind of responses were you getting from your tests. 
- What kind of logic would solve the problems regardless of if we know how to implement it? 

When you explain, in detail, where you are going and how you want to get there it makes your life a lot easier. 

You are amazing at explaining things and I'd like see a write up on how to solve this coding solution in plain language. **Next week I will be looking for pseudo code that teaches me how to code out the problem.** I won't be looking for the code itself to be perfect. 

Let's play to your strengths! This PR is approved!

**Comment by alappies** [comment]
Thanks so much Austin!

Sent from my iPhone

On Sep 28, 2021, at 10:56 AM, Austin Walker ***@***.***> wrote:

﻿

@WalkerTexas-coder approved this pull request.

Ashley, you have some brilliant interview answers and your react debugging is on point! Where I feel this assessment missed on an opportunity was in your pseudo code. You have some solid logic in your code but without the communication that explains where you're headed and whats going on with the code its almost impossible to know when you've gotten what you want.

For instance:

  *   Did any of your tests pass?
  *   What kind of responses were you getting from your tests.
  *   What kind of logic would solve the problems regardless of if we know how to implement it?

When you explain, in detail, where you are going and how you want to get there it makes your life a lot easier.

You are amazing at explaining things and I'd like see a write up on how to solve this coding solution in plain language. Next week I will be looking for pseudo code that teaches me how to code out the problem. I won't be looking for the code itself to be perfect.

Let's play to your strengths! This PR is approved!

________________________________

In code-challenges.test.js<https://github.com/learn-academy-2021-delta/week-3-assessment-alappies/pull/1#discussion_r717814672>:

>  // b) Create the function that makes the test pass.
+//Psuedo Coding
+//First I need to create a function
+// I need the function to take in a number and return it as an array
+//I need the array to spit out the fibonacci sequence starting from 2 with the end point being the number I input.


this is a good start, explain "How" you want this to happen.

________________________________

In code-challenges.test.js<https://github.com/learn-academy-2021-delta/week-3-assessment-alappies/pull/1#discussion_r717816978>:

>
+// var num1 = 1
+// const fibonacciMe = (input) =>{
+//     let newArr = []
+//     for (let index = 0; index < input.length; index++) {
+//     return newArr((num1 -1) + (num1 -2))
+//     }
+//     return newArr.push(fibonacciMe)
+
+// }


Okay both of these solutions are close!

  *   What issues did you run into?
  *   What responses did you get?
  *   Did you get the tests to pass?

This is all necessary information if others are going to be able to help you.

________________________________

In code-challenges.test.js<https://github.com/learn-academy-2021-delta/week-3-assessment-alappies/pull/1#discussion_r717819287>:

>
 // b) Create the function that makes the test pass.
-
-
+//Pseudo Code:
+// I need a function that takes in arrays  const sortThisOut  = [] ==>{}
+// I need that function to return only numbers using typeOf()


This is great! why didn't you implement this?

________________________________

In code-challenges.test.js<https://github.com/learn-academy-2021-delta/week-3-assessment-alappies/pull/1#discussion_r717820242>:

> +// const sortThisOut = (array) =>{
+// let newArray = []
+// for (let index = 0; index < array.length; index++) {
+//     if(array % 2 === 1){
+//         newArray.push(array)
+//     }
+// }
+// return newArray.typeOf(Number)
+// }
+// // console.log(sortThisOut(fullArr2));
+
+describe(sortThisOut, ()=> {
+    it('takes in an array and returns a new array of only odd numbers sorted from least to greatest', ()=> {
+    expect(sortThisOut(fullArr1)).toEqual([-9, 7, 9, 199])
+    })
+    })


Lets focus on communicating those responses and test results! You got this!

________________________________

In code-challenges.test.js<https://github.com/learn-academy-2021-delta/week-3-assessment-alappies/pull/1#discussion_r717823371>:

> +        const element = array[index];
+    }
+    return array.push(iCannot)
+}
+
+describe(iCannot, ()=> {
+    it('is function that takes in an array and returns an array of the accumulating sum', ()=> {
+    expect(iCannot(numbersToAdd1)).toEqual([2, 6, 51, 60])
+    })
+    })
+
+// //Psuedo Codding:
+// First I need to make a function that takes in an array
+// I need to much like the fibonacci sequence have the array go through a for loop
+// I need to iterate the numbers in the array through an equation that will add the current number to the number of the previous index.
+// I need it to return in the form of an array.


Okay I can see here you mostly copied the fib sequence math. That can sometimes throw us off really hard as developers. Pseudo coding this problem out in your own words and writing down all the logical steps will make this problem a lot easier on yourself.

________________________________

In interview-questions.md<https://github.com/learn-academy-2021-delta/week-3-assessment-alappies/pull/1#discussion_r717824799>:

> @@ -10,58 +10,72 @@ Then, research the question to expand on your answer. Even if you feel you have
 1. What is `this` in JavaScript?

   Your answer:
+  The relavance of `this` is apparent in the Javacript method of "class." You would use "this" when refering to something you want to add a value to in your class. example: this.gas = 10.


This is a solid answer!

________________________________

In interview-questions.md<https://github.com/learn-academy-2021-delta/week-3-assessment-alappies/pull/1#discussion_r717825152>:

>

 2. What is React? Why would you use it?

   Your answer:
+  This I feel like I can confidently answer without much research. React is a mulitfacited open-source tool you can use to build user interfaces such as webpages and apps. It can be used to build decadent websites like "Netflix" and detailed apps like facebook and twitter. I would use React because it is one of the leading industry tools in app developement today and once you get used to the JSX language, it can be pretty straight forward to use (realize I did not say easy).


Amazing answer!

________________________________

In interview-questions.md<https://github.com/learn-academy-2021-delta/week-3-assessment-alappies/pull/1#discussion_r717825628>:

>
   Researched answer:
-
+The constructor is the method required for a react class component.


Almost, it's the render() method! Honestly would've gotten me on this one too.

________________________________

In interview-questions.md<https://github.com/learn-academy-2021-delta/week-3-assessment-alappies/pull/1#discussion_r717826526>:

>
   Researched answer:
-
+Ok so this was a trick question. "yarn" is a javascript packaging manager that makes up a react app, so when you run yarn you open up a browser address called "local host 3000" and you're hosting your own react website/app.


correct everything that we use in terminal with a word in the front (yarn, git, node) all reefers to a software that we are calling commands from.

________________________________

In react-debugging.html<https://github.com/learn-academy-2021-delta/week-3-assessment-alappies/pull/1#discussion_r717826726>:

> @@ -79,6 +77,7 @@ <h1>Draw a Card</h1>
                   )
                 })}
               </div>
+              </>


Great work!

**Pull request merged**